cni/pkg/nodeagent/server.go

```diff
@@ -157,16 +157,20 @@ func (s *Server) Start() {
 	log.Info("CNI ambient server starting")
 	s.kubeClient.RunAndWait(s.ctx.Done())
 	log.Info("CNI ambient server kubeclient started")
+	// Start the informer handlers FIRST, before we snapshot.
```

Comment:
Actually looking at this, L158 does the same check.

Not against the change, it just doesn't seem like we had a bug before were the L163 GetActiveAmbientPodSnapshot would incomplete... right? Or am I missing something

Reply:
I seriously doubt there is a bug here yea.

What I am trying to guarantee here is that it is impossible for there to be any window where 

1. `dataplane` starts (allowing a ztunnel to connect)
2. we start sending a snapshot (with pod adds) as part of that connect
3. before the handlers are synced (in case that causes us to lose a delete happening to a pod in the snapshot, or causes us to get a delete for a pod _before_ we send the snapshot `pod add` for that pod).

I'm not sure this was possible with the old order but it's definitely impossible with the new order - we won't take a zt connection until both are started/synced/ready (and we won't mark ourselves as ready until then either, which is currently mostly cosmetic, but certainly more correct).